grubby.c

```diff
@@ -1918,6 +1920,23 @@ static int writeConfig(struct grubConfig *cfg, char *outName,
 			unlink(outName);
 			return 1;
 		}
+
+		// reopen the configuration file for flushing
+		int outfd = open(outName, O_RDONLY);
+		// open both dirname(outName) and dirname(tmpOutName) and pass both fds to fsync()
+		int dirfd1 = open(dirname(strdupa(outName)), O_RDONLY);
+		int dirfd2 = open(dirname(strdupa(tmpOutName)), O_RDONLY);
```

Comment:
In case both files are in the same directory, you can skip one set of the `open()` and `fsync()` calls.